exp/api/v1beta1/azuremachinepool_webhook.go

```diff
@@ -72,15 +80,6 @@ func (amp *AzureMachinePool) ValidateDelete() error {
 
 // Validate the Azure Machine Pool and return an aggregate error.
 func (amp *AzureMachinePool) Validate(old runtime.Object) error {
-	// NOTE: AzureMachinePool is behind MachinePool feature gate flag; the web hook
```

Comment:
Don't we still want to exit early here in `Validate()` if MachinePools aren't enabled?

Reply:
I made this change because it makes the most sense to throw this error during resource create time.